Adds extra elements to RSS items. (#6707)
withastro/astro: packages/astro-rss/README.md

  content?: string;
  /** Append some other XML-valid data to this item */
  customData?: string;
  /** Categories or tags related to the item */

>   categories?: string[];

Reviewer: Maybe we could provide some examples of these categories?

  /** The item author's email address */
  author?: string;
  /** A URL of a page for comments relating to the item */
  comments?: string;
  /** The RSS channel that the item came from */

>   source?: { url: string, title: string }

Reviewer: We should explain the fields. It's not clear, for example, what `title` is

  /** A media object that belongs to the item */

>   enclosure?: { url: string, length: number, type: string }

Reviewer: We should explain/document all these fields. It's not clear what `length` and `type` are

});
```

> ## `RSSFeedItem`

Reviewer 2: I really like how well you've restructured the docs and how well you explained this

An `RSSFeedItem` is a single item in the list of items in your feed. It represents a story, with `link`, `title` and `pubDate` fields. There are further optional fields defined below. You can also check the definitions for the fields in the [RSS spec](https://validator.w3.org/feed/docs/rss2.html#ltpubdategtSubelementOfLtitemgt).

An example feed item might look like:

```js
const item = {
  title: "Alpha Centauri: so close you can touch it",
  link: "/blog/alpha-centuari",
  pubDate: new Date("2023-06-04"),
  description: "Alpha Centauri is a triple star system, containing Proxima Centauri, the closest star to our sun at only 4.24 light-years away.",
  categories: ["stars", "space"]
}
```

### `title`

Type: `string (required)`

The `<title>` attribute of the item in the feed.

### `link`

Type: `string (required)`

The `<link>` attribute of the item in the feed containing the URL of the item on the web.

### `pubDate`

Type: `Date (required)`

Indicates when the item was published.

### `description`

Type: `string (optional)`

> If the item is complete within itself, that is you are publishing the full content of the item in the feed, the `description` field may contain the full text (entity-encoded HTML is permitted). If the item is a stub, then the `description` may contain a synopsis of the item.

Author: Yeah, RSS is old and a mess! And feed readers have inconsistent implementations. Such fun!

### `content`

Type: `string (optional)`

If you want to supply both a short description and also the full content in an item, set the `content` field to the full, encoded text. See the [recommendations from the RSS spec for how to use and differentiate between `description` and `content`](https://www.rssboard.org/rss-profile#namespace-elements-content-encoded).

### `categories`

Type: `string[] (optional)`

If you use tags or categories to categorize your content, you can add them as the `categories` field. They will be output as multiple `<category>` elements.

### `author`

Type: `string (optional)`

> Useful for multi-author blogs, the `author` field provides the email address of the person who wrote the item.

Author: Agreed, but that's what the spec says. 🤷‍♂️

### `commentsUrl`

Type: `string (optional)`

The `commentsUrl` defines a URL of a web page that contains comments on the item.

### `source`

Type: `object (optional)`

> Items that are republished from other publications may define a `source` which defines the `title` and `url` of the original feed in which it was published.

Reviewer 3: ``````suggestion
An object that defines the `title` and `url` of the original feed for items that have been republished from another source. Both are required propeties of `source` for proper attribution. 

```js
const item = {
	title: "Alpha Centauri: so close you can touch it",
  link: "/blog/alpha-centuari",
  pubDate: new Date("2023-06-04"),
	source:
		{
			title: "The Galactic Times",
			url: "https://galactictimes.space/feed.xml"
		}
}
```
``````
I also think an example (choose one that makes sense, formatted properly!) would make sense here.

> #### `title`

Reviewer 3: ```suggestion
#### `source.title`
```
I'm wondering if this is helpful, especially as a heading that might show up in Algolia if someone searches for `title` in another context, like frontmatter of a Markdown file. Also helps to connect the properties to their object.

Author: Good idea, thanks.

Type: `string (required)`

> If you define a `source` you must define that source's `title`. It is the name of the original feed in which the item was published.

Reviewer 3: ```suggestion
The name of the original feed in which the item was published. (Note that this is the the feed's title, not the individual article title.)
```

> #### `url`

Reviewer 3: ```suggestion
#### `source.url`
```

Type: `string (required)`

> If you define a `source` you must also define that source's `url` which identifies the URL of the original feed in which the item was published.

Reviewer 3: ```suggestion
The URL of the original feed in which the item was published.
```

### `enclosure`

Type: `object (optional)`

> Items that include media as part of the feed, like a podcast, can define an `enclosure` which is made of three required fields, a `url`, `length`, and `type`.

Reviewer 3: ``````suggestion
An object to specify properties for an included media source (e.g. a podcast) with three required values: `url`, `length`, and `type`.

```js
const item = {
	title: "Alpha Centauri: so close you can touch it",
  link: "/blog/alpha-centuari",
  pubDate: new Date("2023-06-04"),
	enclosure:
		{
			url: "https://mydomain/media/alpha-centauri.aac",
			length: 124568,
			type: "audio/aac"
		}
}
```
``````
I also think an example would be helpful here! Note, if the URL is from your own site, do you still need to use a full URL here? This is just an example. Adjust to make this correct!

Author: The RSS spec says that you do need this to be a full URL. However, we can treat this like `link`, check if it's a valid URL and if it isn't run it through `createCanonicalURL` to make a full URL for a relative path.

> #### `url`

Reviewer 3: ```suggestion
#### `enclosure.url`
```

Type: `string (required)`

> The `url` field for the `enclosure` defines a URL where the media can be found.

Reviewer 3: ```suggestion
The URL where the media can be found.
```
If you only need to use e.g. `/media/filename` for media hosted on your own site, and not the full URL, then I would also add a sentence mentioning that full URLs are only required for external sources. (And, make sure the example code in `enclosure` demonstrates a correct URL!)

> #### `length`

Reviewer 3: ```suggestion
#### `enclosure.length`
```

Type: `number (required)`

> The `length` field defines the size of the file found at the `url` in bytes.

Reviewer 3: ```suggestion
The size of the file found at the `url` in bytes.
```

> #### `type`

Reviewer 3: ```suggestion
#### `enclosure.type`
```

Type: `string (required)`

> The `type` field defines the MIME type for the media item found at the `url`.

Reviewer 3: ```suggestion
The [MIME type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types) for the media item found at the `url`.
```
